cli/repl.rs

```diff
@@ -32,6 +38,82 @@ impl Validator for Helper {
   }
 }
 
+impl Highlighter for Helper {
+  fn highlight_hint<'h>(&self, hint: &'h str) -> Cow<'h, str> {
+    hint.into()
+  }
+
+  fn highlight<'l>(&self, line: &'l str, pos: usize) -> Cow<'l, str> {
+    self.highlighter.highlight(line, pos)
+  }
+
+  fn highlight_candidate<'c>(
+    &self,
+    candidate: &'c str,
+    _completion: rustyline::CompletionType,
+  ) -> Cow<'c, str> {
+    self.highlighter.highlight(candidate, 0)
+  }
+
+  fn highlight_char(&self, line: &str, _: usize) -> bool {
+    !line.is_empty()
+  }
+}
+
+struct LineHighlighter {
+  regex: Regex,
+}
+
+impl LineHighlighter {
+  fn new() -> Self {
+    let regex = Regex::new(
+      r#"(?x)
+      (?P<comment>(?:/\*[\s\S]*?\*/|//[^\n]*)) |
+      (?P<string>(?:"([^"\\]|\\.)*"|'([^'\\]|\\.)*'|`([^`\\]|\\.)*`)) |
+      (?P<regexp>/(?:(?:\\/|[^\n/]))*?/[gimsuy]*) |
+      (?P<number>\d+(?:\.\d+)*(?:e[+-]?\d+)*n?) |
+      (?P<boolean>\b(?:true|false)\b) |
+      (?P<null>\b(?:null)\b) |
+      (?P<undefined>\b(?:undefined)\b) |
+      (?P<keyword>\b(?:await|async|var|let|for|if|else|in|of|class|const|function|yield|return|with|case|break|switch|import|export|new|while|do|throw|catch)\b) |
```

Comment:
missing `this`

Reply:
Indeed, treating it as identifier (no color) for now.